src/neo/Network/P2P/Payloads/Header.cs

```diff
@@ -144,6 +155,7 @@ void IVerifiable.DeserializeUnsigned(BinaryReader reader)
             merkleRoot = reader.ReadSerializable<UInt256>();
             timestamp = reader.ReadUInt64();
             index = reader.ReadUInt32();
+            nonce = reader.ReadUInt32();
```

Comment:
I agree with it as well.
Another possibility would be to create some Shamir Secret between all consensus nodes and share it just on the P2P and, then, feed this value by oracle on the next block.

Reply:
> We're at RC2, changing block header at this stage doesn't look nice to me.

Agree, this will make a hard fork, and I am working on moving it to a transaction. The plan is, allowing the primary to generate a 0 fee `transaction` to store the `nonce` and attach it to the end of the `Transaction list`.